Mathlib/Logic/Embedding/Set.lean

```diff
@@ -147,3 +152,55 @@ theorem subtypeOrEquiv_symm_inr (p q : α → Prop) [DecidablePred p] (h : Disjo
 #align subtype_or_equiv_symm_inr subtypeOrEquiv_symm_inr
 
 end Subtype
+
+section Disjoint
+
+variable {α ι : Type*} {s t r : Set α}
+
+/-- For disjoint `s t : Set α`, the natural injection from `↑s ⊕ ↑t` to `α`. -/
+@[simps] def Disjoint.sumSubtypeEmbedding (h : Disjoint s t) : s ⊕ t ↪ α where
+  toFun := Sum.elim (↑) (↑)
+  inj' := by
+    rintro (⟨a, ha⟩ | ⟨a, ha⟩) (⟨b, hb⟩ | ⟨b, hb⟩)
+    · simp [Subtype.val_inj]
+    · simpa using h.ne_of_mem ha hb
+    · simpa using h.symm.ne_of_mem ha hb
+    simp [Subtype.val_inj]
```

Comment:
This could equivalently be written as
```suggestion
@[simps] def Disjoint.sumSubtypeEmbedding (h : Disjoint s t) : s ⊕ t ↪ α :=
  (Equiv.Set.union h).toEmbedding.trans (.subtype _)
```
(except for the fact that `Equiv.Set.union` takes in `s ∩ t ⊆ ∅` rather than `Disjoint s t`, which should be fixed), so maybe not worth a separate definition?

Reply:
This has the advantage of not needing decidability. Also, the simp lemmas below would look quite clunky without a definition.

Reply:
> the simp lemmas below would look quite clunky without a definition

That's usually not a concern since there are simp lemmas dealing with the intermediate expression




> This has the advantage of not needing decidability

That is a fair point

Reply:
Please show the equality with my spelling, though